File: 5/5-297_classic.pas
// Автор В.Н. Шубинкин

function control_sum(num: BigInteger): BigInteger;
begin
  while num > 0 do
  begin
    result += num mod 10;
    num := num div 10;
    var d := num mod 10 * 2;
    result += d div 10 + d mod 10;
    num := num div 10
  end;
end;

begin
  var num := 1000000000000000;
  num += 1;
  while control_sum(num) <> 25 do num += 1;
  print(num mod 10bi**15)
end.

File: Dima/5/5-297_classic.py
# Автор В.Н. Шубинкин

def control_sum(num):
    s = 0
    while num:
        s += num % 10
        num //= 10
        d = 2 * (num % 10)
        d = d % 10 + d // 10
        s += d
        num //= 10
    return s


num = 1000_0000_0000_0000
num += 1
while control_sum(num) != 25:
    num += 1
print(num, num % 10**15)
